### ISSUE TITLE
Summary of suggestions for improvements to the builder

### ISSUE THREAD

**Issue by tpiperatgod** [OpenFunction/builder#24]
As we mentioned in the readme, the builder project is based on [gcp/buildpacks](https://github.com/GoogleCloudPlatform/buildpacks) with some modifications to facilitate support for Chinese users and support for OpenFunction's functions-framework. [gcp/buildpacks](https://github.com/GoogleCloudPlatform/buildpacks) primarily serves GCP, so we may need to do more work on improving builder customization.

Of course, we are also considering rewriting a `Builder` project that will still follow the [buildpacks lifecycle](https://github.com/buildpacks/lifecycle) criteria (which drove us to find out more about the builder's improvement needs).

**Comment by tpiperatgod** [comment]
## Feature request
We need to enable builder to support custom configuration of source code repository addresses, dependency repository addresses, etc.

## User case
This would make sense in the case of using a local(or private) repository.

**Comment by benjaminhuo** [comment]
> As we mentioned in the readme, the builder project is based on [gcp/buildpacks](https://github.com/GoogleCloudPlatform/buildpacks) with some modifications to facilitate support for Chinese users and support for OpenFunction's functions-framework. [gcp/buildpacks](https://github.com/GoogleCloudPlatform/buildpacks) primarily serves GCP, so we may need to do more work on improving builder customization.
> 
> Of course, we are also considering rewriting a `Builder` project that will still follow the [buildpacks lifecycle](https://github.com/buildpacks/lifecycle) criteria (which drove us to find out more about the builder's improvement needs).

Agree, we should have our own builder and should have a better understanding to the buildpacks tech

**Comment by tpiperatgod** [comment]
## Feature request
Pass through the information (package name, version number, etc.) of **functions-framework** via environment variables.

## User case
This will decouple the **builder** with the **functions-framework**.
